src/datachain/dataset.py

```diff
@@ -358,10 +388,39 @@ def parse_schema(
             for c_name, c_type in ct.items()
         }
 
+    @staticmethod
+    def validate_name(name: str) -> None:
+        """Throws exception if name has reserved characters"""
+        for c in DATASET_NAME_RESERVED_CHARS:
+            if c in name:
+                raise InvalidDatasetNameError(
+                    f"Character {c} is reserved and not allowed in dataset name"
+                )
+
+    @staticmethod
+    def clean_name(name: str) -> str:
```

Comment:
so, do we validate or replace? (not sure when we one thing and when the second one)